src/Nest/Modules/SnapshotAndRestore/Repositories/S3Repository.cs

```diff
@@ -7,131 +7,271 @@ public interface IS3Repository : IRepository<IS3RepositorySettings> { }
 
 	public class S3Repository : IS3Repository
 	{
-		public S3Repository(S3RepositorySettings settings)
-		{
-			Settings = settings;
-		}
+		public S3Repository(S3RepositorySettings settings) => Settings = settings;
 
 		public IS3RepositorySettings Settings { get; set; }
 		public string Type { get; } = "s3";
 	}
 
 	public interface IS3RepositorySettings : IRepositorySettings
 	{
+		/// <summary>
+		/// The name of the bucket to be used for snapshots. This field is required
+		/// </summary>
 		[JsonProperty("bucket")]
 		string Bucket { get; set; }
 
+		/// <summary>
+		/// The region where bucket is located. Defaults to US Standard.
+		/// </summary>
 		[Obsolete("Removed in NEST 6.x.")]
 		[JsonProperty("region")]
 		string Region { get; set; }
 
+		/// <summary>
+		/// The endpoint to the S3 API. Defaults to AWS’s default S3 endpoint. Note that setting a region overrides the endpoint setting.
+		/// </summary>
+		[Obsolete("Removed in NEST 6.x, but can be specified on client settings in elasticsearch.yml configuration.")]
+		[JsonProperty("endpoint")]
+		string Endpoint { get; set; }
+
+		/// <summary>
+		/// The protocol to use (http or https). Defaults to value of cloud.aws.protocol or cloud.aws.s3.protocol.
+		/// </summary>
+		[Obsolete("Removed in NEST 6.x, but can be specified on client settings in elasticsearch.yml configuration.")]
+		[JsonProperty("protocol")]
+		string Protocol { get; set; }
+
+		/// <summary>
+		/// Specifies the path within bucket to repository data.
+		/// Defaults to value of repositories.s3.base_path or to root directory if not set.
+		/// </summary>
 		[JsonProperty("base_path")]
 		string BasePath { get; set; }
 
+		/// <summary>
+		/// The access key to use for authentication. Defaults to value of cloud.aws.access_key.
+		/// </summary>
 		[JsonProperty("access_key")]
 		string AccessKey { get; set; }
 
+		/// <summary>
+		/// The secret key to use for authentication. Defaults to value of cloud.aws.secret_key.
```

Comment:
add `from elasticsearch.yml configuration` to default description

Reply:
:+1: